### PR TITLE
New TextFirst flag for syntax highlight

### PR DESCRIPTION
**Pull request by anderson-joyle**
Issue https://github.com/microsoft/Power-Fx/issues/2161

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Lexer/ITextFirstFlag.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.PowerFx.Syntax
+{
+    /// <summary>
+    /// Indicates that the object is in active TextFirst context.
+    /// </summary>
+    public interface ITextFirstFlag
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Lexer/ITextFirstFlag.cs:13]: >public [](http://example.com/codeflow?start=4&length=6)

let's remove interface and just keep on STrLitToken directly

File: src/tests/Microsoft.PowerFx.Core.Tests/LexerTests.cs [hunk 2]
```diff
@@ -605,5 +605,30 @@ public void TextFirstBasicFormulaStrLit()
             Assert.True(lit.Span.Min == 1);
             Assert.True(lit.Span.Lim == str.Length);
         }
+
+        [Theory]
+        [InlineData("Hello ${\"Word\"}", 0, 1, 5)]
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/LexerTests.cs:610]: >0, 1, 5 [](http://example.com/codeflow?start=41&length=7)

per discussion, there are some tricks to make this test more maintainable

**Review by MikeStall — DISMISSED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**